cmake/FindGeant4.cmake

```diff
@@ -1,25 +1,31 @@
 # Find Geant4 cmake config file
 set(Geant4_SEARCH_DIRS)
-# Is this line necessary?
-set(PACKAGE_FIND_VERSION ${Geant4_FIND_VERSION})
+set(Geant4_FIND_VERSION ${Geant4_FIND_VERSION})
 file(GLOB Geant4_SEARCH_DIRS "${GEANT4_DIR}/lib*/Geant4-*")
 string(REPLACE "\n" ";" Geant4_SEARCH_DIRS ${Geant4_SEARCH_DIRS})
 
-find_path(Geant4_CMAKE_CONFIG_VERSION
-  NAMES Geant4ConfigVersion.cmake
-  PATHS ${Geant4_SEARCH_DIRS}
-  NO_DEFAULT_PATH
-  )
+# Extract the version number from the first directory in Geant4_SEARCH_DIRS
+if(Geant4_SEARCH_DIRS)
+    # Extract the version number from the directory name
+    list(GET Geant4_SEARCH_DIRS 0 Geant4_LIB_DIR)
+    file(RELATIVE_PATH Geant4_VERSION ${GEANT4_DIR} ${Geant4_LIB_DIR})
+    string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+" Geant4_VERSION ${Geant4_VERSION})
 
-if (Geant4_CMAKE_CONFIG_VERSION)
-  set(Geant4_CMAKE_CONFIG_VERSION ${Geant4_CMAKE_CONFIG_VERSION}/Geant4ConfigVersion.cmake)
```

Comment:
Does Geant no longer provide a cmake file that we can interrogate?  That seems preferable to parsing a string based on a directory name....

Reply:
I'm unsure whether `Geant4` provides and maintains this file within its distribution. I have developed a method to extract the version of the directory structure itself. The advantage is that it can work even if `Geant4` doesn't provide a version-specific configuration file. It directly extracts version information from the installation path. However, the downside is that it's vulnerable to changes in the directory structure, which could potentially lead to parsing errors. It relies on consistent naming conventions.

Reply:
One of the values of CMake is that is automates this in a robust way.  I'll take a look to see what I find in a new distribution

Reply:
Yes. `Geant4` provides and maintains this file within its distribution.

I am testing DAGMC with following arguments and its working. Let's see if it passes the workflow.

```cmake
  cmake ../ -DMOAB_CMAKE_CONFIG=$env_dir/lib/cmake/MOAB \
    -DMOAB_DIR=$env_dir \
    -DBUILD_STATIC_LIBS=OFF \
    -DBUILD_GEANT4=ON \
    -DGeant4_CMAKE_CONFIG=$env_dir/lib/cmake/Geant4 \
    -DGEANT4_DIR=$env_dir \
    -DBUILD_TALLY=ON \
    -DCMAKE_INSTALL_PREFIX=$env_dir
```